tests/models/nn/test_bertrec.py

```diff
@@ -0,0 +1,264 @@
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import typing as tp
+
+import numpy as np
+import pandas as pd
+import pytest
+import torch
+from pytorch_lightning import seed_everything
+
+from rectools import Columns
+from rectools.models import BERT4RecModel
+from rectools.models.nn.bert4rec import BERT4RecDataPreparator
+from rectools.models.nn.item_net import IdEmbeddingsItemNet
+from rectools.models.nn.transformer_base import (
+    LearnableInversePositionalEncoding,
+    PreLNTransformerLayers,
+    SessionEncoderLightningModule,
+)
+from tests.models.data import DATASET
+from tests.models.utils import assert_default_config_and_default_model_params_are_the_same
+
+
+def leave_one_out_mask(interactions: pd.DataFrame) -> pd.Series:
+    rank = (
+        interactions.sort_values(Columns.Datetime, ascending=False, kind="stable")
+        .groupby(Columns.User, sort=False)
+        .cumcount()
+    )
+    return rank == 0
+
+
+class TestBERT4RecModelConfiguration:
+    def setup_method(self) -> None:
+        self._seed_everything()
+
+    def _seed_everything(self) -> None:
+        torch.use_deterministic_algorithms(True)
+        seed_everything(32, workers=True)
+
+    def test_from_config(self) -> None:
+
+        config = {
+            "n_blocks": 2,
+            "n_heads": 4,
+            "n_factors": 64,
+            "use_pos_emb": False,
+            "use_causal_attn": False,
+            "use_key_padding_mask": True,
+            "dropout_rate": 0.5,
+            "session_max_len": 10,
+            "dataloader_num_workers": 5,
+            "batch_size": 1024,
+            "loss": "softmax",
+            "n_negatives": 10,
+            "gbce_t": 0.5,
+            "lr": 0.001,
+            "epochs": 10,
+            "verbose": 1,
+            "deterministic": True,
+            "recommend_accelerator": "auto",
+            "recommend_devices": 1,
+            "recommend_batch_size": 256,
+            "recommend_n_threads": 0,
+            "recommend_use_gpu_ranking": True,
+            "train_min_user_interactions": 5,
+            "item_net_block_types": (IdEmbeddingsItemNet,),
+            "pos_encoding_type": LearnableInversePositionalEncoding,
+            "transformer_layers_type": PreLNTransformerLayers,
+            "data_preparator_type": BERT4RecDataPreparator,
+            "lightning_module_type": SessionEncoderLightningModule,
+            "mask_prob": 0.15,
+            "get_val_mask_func": leave_one_out_mask,
+        }
+        model = BERT4RecModel.from_config(config)
+        assert model.n_blocks == 2
```

Comment:
I think here we can safely reduce the code with
```
for key, config_value in config.items():
    assert getattr(model, key) == config_value
```

Reply:
Same or similar for other tests

Reply:
fixed